sdk/kiosk/src/client/tp-manager.ts

```diff
@@ -0,0 +1,422 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { TransactionArgument, type TransactionBlock } from '@mysten/sui.js/transactions';
+import {
+	attachFloorPriceRuleTx,
+	attachKioskLockRuleTx,
+	attachPersonalKioskRuleTx,
+	attachRoyaltyRuleTx,
+} from '../tx/rules/attach';
+import {
+	createTransferPolicy,
+	createTransferPolicyWithoutSharing,
+	removeTransferPolicyRule,
+	shareTransferPolicy,
+	withdrawFromPolicy,
+} from '../tx/transfer-policy';
+import {
+	type KioskClientOptions,
+	type Network,
+	type ObjectArgument,
+	TRANSFER_POLICY_CAP_TYPE,
+} from '../types';
+import { queryOwnedTransferPolicyCap, queryTransferPolicy } from '../query/transfer-policy';
+import { SuiClient } from '@mysten/sui.js/src/client';
+import {
+	FLOOR_PRICE_RULE_ADDRESS,
+	KIOSK_LOCK_RULE_ADDRESS,
+	PERSONAL_KIOSK_RULE_ADDRESS,
+	ROYALTY_RULE_ADDRESS,
+} from '../constants';
+
+export type TransferPolicyBaseParams = {
+	type: string;
+	publisher: ObjectArgument;
+	skipCheck?: boolean;
+};
+
+export class TransferPolicyManager {
+	client: SuiClient;
+	network: Network;
+	policyId?: ObjectArgument;
```

Comment:
the way this manager is stateful feels a little dangerous.  What you could do instead is make methods like `create` return a new instance with the these properties set (or maybe better would be to have them be static methods on the class).

I think the issue here is that this is mixing methods that apply to a kiosk/collection of policies with methods for a specific kiosk in a stateful way.    Having to set and reset the policy/cap in various methods is a bit of a red flag, and I think it would be easy to accidentally end up in a bad state, or in a situation where you are mutating these things in parallel for async methods.

Reply:
I think we'll probably proceed with a similar approach to `KioskTransaction` on this one too. Since we also have the packageId overrides too, this makes a lot of sense.